vllm/core/block/block_table.py

```diff
@@ -65,6 +65,26 @@ def get_num_required_blocks(token_ids: List[int], block_size: int) -> int:
         """
         return cdiv(len(token_ids), block_size)
 
+    def append_by_blocks(self,
```

Comment:
This is pretty similar to the normal allocation -- let's try to avoid introducing new APIs if we can

Reply:
Sure I can merge it with the original `allocate` function, PTAL. But a nicht is that normal `allocate` does not need to return the blocks it populate, but for swapping, since we need to record the `block_id` of the new block, we need to return such info.

Reply:
We can use this pattern: 
![image](https://github.com/vllm-project/vllm/assets/950914/cc7c648c-3201-404a-aaf6-ab0d99c9fc67)

The scheduler would then retrieve the swaps at the end of scheduling.

Reply:
Refactor the swapping logic following this pattern. PTAL!